src/pages/en/guides/client-side-scripts.md

```diff
@@ -0,0 +1,205 @@
+
+```astro title="src/components/Example.astro"
+<script>
+  // Processed! Bundled! TypeScript-supported!
+  // ESM imports work, even to npm packages.
+</script>
+```
+
+To avoid bundling the script, you can use the `is:inline` attribute.
+
+```astro title="src/components/InlineScript.astro" "is:inline"
+<script is:inline>
+  // Will be rendered into the HTML exactly as written!
+  // ESM imports will not be resolved relative to the file.
+</script>
+```
+
+:::note
+Adding `type="module"` or any other attribute to a `<script>` tag will disable Astro's default bundling behavior, treating the tag as if it had an `is:inline` directive.
+:::
+
+📚 See our [directives reference](/en/reference/directives-reference/#script--style-directives) page for more information about the directives available on `<script>` tags.
+
+### Script loading
+
+You may want to write your scripts as separate `.js`/`.ts` files or need to reference an external script on another server. You can do this by referencing these in a `<script>` tag’s `src` attribute.
+
+#### Import local scripts
+
+**When to use this:** If your script lives inside of `src/`.
+
+Astro will build, optimize, and add these scripts to the page for you, following its [script bundling rules](#script-bundling).
+
+```astro title="src/components/LocalScripts.astro"
+<!-- relative path to script in `src/` -->
+<script src="./local-script.js"></script>
+
+<!-- also works for local TypeScript files -->
+<script src="./script-with-types.ts"></script>
+```
+
+#### Load external scripts
+
+**When to use this:** If your JavaScript file lives inside of `public/` or on a CDN.
+
+This approach skips the JavaScript processing, bundling, and optimizations that are provided by Astro when you import scripts as described above.
+
+```astro title="src/components/ExternalScripts.astro" "is:inline"
+<!-- absolute path to a script at `public/my-script.js` in your project -->
+<script is:inline src="/my-script.js"></script>
+
+<!-- full URL to a script on a remote server -->
+<script is:inline src="https://analytics.example.com/script.js"></script>
+```
+
+## Common script patterns
+
+### Handle `onclick` and other events
+
+Some UI frameworks use custom syntax for event handling like `onclick={...}` (React/Preact) or `@click="..."` (Vue). Astro components follow web standards and recommend using [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) in a `<script>` tag to handle user interactions.
+
+```astro title="src/components/AlertButton.astro"
+<button data-alert>Click me!</button>
+
+<script>
+  // Find all buttons with the `data-alert` attribute on the page.
+  const buttons = document.querySelectorAll('button[data-alert]');
+
+  // Handle clicks on each button.
+  buttons.forEach((button) => {
+    button.addEventListener('click', () => {
+      alert('Button was clicked!');
+    });
+  });
+</script>
+```
```

Comment:
If this example works with a css `class=""` instead of a `data-` attribute, I'm *tempted* to show that here instead. (Not opposed to what's here, just thinking...)

My thinking: 
- CSS classes might be more familiar, and might already be in use for an element anyway.
- You have a nice introduction/explanation to `data-`  attributes in the "using frontmatter variables" section. So, it seems weird to be using them here, then introducing them at the bottom of the page.

We *could* move that introduction up here,  but I think this is a simple enough (and often searched!) example that we don't need to clutter what's not very complicated here.

In which case I might even add to the line above with something like:

> You can add [`addEventListeners`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) to elements identified by class, id, attribute  etc. in a `<script>` tag to handle user interactions.

Reply:
Agree — `data-*` attributes are often recommended for this because they help you separate styling and behaviour, but `class` examples are more common, and showing the data attribute later means we can cover both bases.

````suggestion
```astro title="src/components/AlertButton.astro"
<button class="alert">Click me!</button>

<script>
  // Find all buttons with the class `alert` attribute on the page.
  const buttons = document.querySelectorAll('button.alert');

  // Handle clicks on each button.
  buttons.forEach((button) => {
    button.addEventListener('click', () => {
      alert('Button was clicked!');
    });
  });
</script>
```
````